src/main.cpp

```diff
@@ -0,0 +1,100 @@
+#include <string>
+#include <tuple>
+#include <map>
+#include <fstream>
+#include <iostream>
+#include <sstream>
+#include <algorithm>
+#include <iterator>
+#include <vector>
+#include <functional>
+#include <regex>
+#include "Coordinate.h"
+#include "IO_util.h"
+
+
+
+std::vector < std::function<double(const Coordinate&, const Coordinate&)>> initializePredicates() {
+	std::vector< std::function<double(const Coordinate&, const Coordinate&)>> distancePredicates = {
+		[](const Coordinate& c1, const Coordinate& c2) {				// Euclidean distance calculation
+			double dx = c2._x - c1._x;
+			double dy = c2._y - c1._y;
+			return std::sqrt(dx * dx + dy * dy);
+		}
+		, [](const Coordinate& c1, const Coordinate& c2) {			// Chebyshev distance calculation
+			double dx = std::abs(c2._x - c1._x);
+			double dy = std::abs(c2._y - c1._y);
+			return std::max(dx, dy);
+		},
+		 [](const Coordinate& c1, const Coordinate& c2) {			// Manhattan distance calculation
+			double dx = std::abs(c2._x - c1._x);
+			double dy = std::abs(c2._y - c1._y);
+			return dx + dy;
+		}
+	};
+	return distancePredicates;
+}
+
+
+
+
+
+int main() {
+	std::map<std::string, Coordinate> city_to_coordinates;
+	std::map<Coordinate, std::string> coordinates_to_city;
+	std::map<Coordinate, std::string> cities_within_radius;
+	std::map<double, std::string> cities_within_radius_by_distance;
+	Coordinate input_city_coordinates;
+
+	std::string city_name;
+	int algorithm_choice = -1;
+	double radius = 0;
+	int  cities_north_to_city_count = 0;
+	auto distancePredicates = initializePredicates();
+	size_t number_of_functions = distancePredicates.size();
+	size_t result_length;
+
+	try {
+		IOUtilities::fillMapsFromFile(city_to_coordinates, coordinates_to_city);
+		city_name = IOUtilities::getCityNameInput(city_to_coordinates);
+
+		do {
+
+
+			radius = IOUtilities::getRadiusInput();
+			algorithm_choice = IOUtilities::getAlgorithmChoiceInput(number_of_functions);
+			input_city_coordinates = city_to_coordinates[city_name];
+
+			std::copy_if(coordinates_to_city.begin(), coordinates_to_city.end(), std::inserter(cities_within_radius, cities_within_radius.end()),
```

Comment:
you are missing 1 step in finding the cities in radius, you should find vertical and horizontal rectangles, and then to find the intersection square, and to find a circle inside the square.